test/test_esapi.py

```diff
@@ -4716,6 +4716,21 @@ def test_ref_parent(self):
         gc.collect()
         self.assertEqual(bytes(algb), TPM2_ALG.SHA256.to_bytes(2, "big"))
 
+    def test_double_close(self):
+        ectx = ESAPI(self.tpm.tcti_name_conf)
+        self.assertTrue(ectx._did_load_tcti)
+        self.assertTrue(bool(ectx._ctx))
+        self.assertTrue(bool(ectx._ctx_pp))
+        self.assertEqual(ectx.tcti.name_conf, self.tpm.tcti_name_conf)
+        ectx.close()
+        self.assertFalse(bool(ectx._ctx))
+        self.assertFalse(bool(ectx._ctx_pp))
+        self.assertEqual(ectx.tcti, None)
+        ectx.close()
+        self.assertFalse(bool(ectx._ctx))
+        self.assertFalse(bool(ectx._ctx_pp))
```

Comment:
You could probably skip bool(...) for assertTrue and assertFalse calls

Reply:
I agree. I was not familiar with `assertTrue`/`assertFalse` but the documentation is clear on https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue : `assertTrue(expr)` checks that `bool(expr) is True`, not that `expr is True`.
Thanks for the review!